services/storage/src/simcore_service_storage/db_file_meta_data.py

```diff
@@ -68,7 +68,7 @@ async def get(conn: SAConnection, file_id: SimcoreS3FileID) -> FileMetaDataAtDB:
 async def list_filter_with_partial_file_id(
     conn: SAConnection,
     *,
-    user_id: UserID,
+    user_id: UserID | None,
```

Comment:
Hmm, yes this is true. When filtering in this function one can definitely pass a filter which no files meet. But this can also be done e.g. with the `sha256_checksum`.

What annoys me is the following: the values for `user_id=None` and `project_ids=[ ]` have a different behavior than the default values of the other parameters because of the or-statement. E.g. if I pass `sha256_checksum=None` then this *extends* the set of files meeting the condition by imposing no restriction on the `sha256_checksum`. On the other hand if I pass `user_id=None` I *restrict* to the set of files only meeting the conditions of the `project_id`s. I would like to at least have names for the special parameters which appear in the `or`-statement to make this clear from outside the function. What do you think about appending all "filter"-parameters (so `file_id_prefix`, `partial_file_id`, `sha256_checksum` and `only_files` by `_filter` to make it clear that these are "real filters" as opposed to the `user_id` and `project_ids` which I don't consider real filters?

Reply:
OK, at second thought I think a reasonable solution is to make clear in the notation that `user_id` and `project_ids` are a single filter. I have done so by introducing a NamedTuple which (hopefully) makes this clear. See https://github.com/ITISFoundation/osparc-simcore/pull/5799/commits/16023b92f697e9420bce4e1f299a4956a6bb0737 and let me know if you agree this is a good strategy